### PR TITLE
Dca 702 custom dataset filtering

### PR DESCRIPTION
**Pull request by SergeyMoroz0703**

### CODE REVIEW

File: app/confluence.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ settings:
   aggregator: consolidator
   verbose: false
   env:
-    application_hostname: test_confluence_instance.atlassian.com   # Confluence DC hostname without protocol and port e.g. test-confluence.atlassian.com or localhost
+    application_hostname:  test_confluence_instance.atlassian.com  # Confluence DC hostname without protocol and port e.g. test-confluence.atlassian.com or localhost
```
Comment (ometelytsia) [inline app/confluence.yml:7]: redundant space

File: app/confluence.yml [hunk 2]
```diff
@@ -14,13 +14,13 @@ settings:
     load_executor: jmeter           # jmeter and locust are supported. jmeter by default.
     concurrency: 200                # number of concurrent virtual users for jmeter or locust scenario
     test_duration: 45m
-    ramp-up: 3m                     # time to spin all concurrent users
+    ramp-up: 5m                     # time to spin all concurrent users
```
Comment (ometelytsia) [inline app/confluence.yml:17]: why ramp-up time was changed?

File: app/confluence.yml [hunk 3]
```diff
@@ -32,6 +32,8 @@ settings:
     upload_attachment: 5
     like_page: 3
     standalone_extension: 0 # By default disabled
+    # Custom dataset section.
+    custom_dataset_query: ''        # Write CQL query to add CQL output to the datasets/confluence/custom-pages.csv
```
Comment (ometelytsia) [inline app/confluence.yml:36]: app/datasets/confluence/custom-pages.csv

Comment (ometelytsia) [inline app/confluence.yml:36]: Are single quotes are needed here?

Comment (ometelytsia) [inline app/confluence.yml:36]: It would be nice to add an example in a comment.

File: app/extension/confluence/extension_ui.py [hunk 4]
```diff
@@ -3,6 +3,8 @@
 from util.conf import CONFLUENCE_SETTINGS
 
 from selenium_ui.base_page import BasePage
+from selenium_ui.confluence.pages.pages import Login, AllUpdates, PopupManager, Page, Dashboard, TopNavPanel, Editor, \
```
Comment (ometelytsia) [inline app/extension/confluence/extension_ui.py:6]: Why this import was added?

File: app/extension/jira/extension_ui.py [hunk 5]
```diff
@@ -3,6 +3,8 @@
 from util.conf import JIRA_SETTINGS
 
 from selenium_ui.base_page import BasePage
+from selenium_ui.jira.pages.pages import Login, PopupManager, Issue, Project, Search, ProjectsList, \
```
Comment (ometelytsia) [inline app/extension/jira/extension_ui.py:6]: Why this import was added?

File: app/jira.yml [hunk 6]
```diff
@@ -34,6 +34,8 @@ settings:
     view_backlog: 6
     browse_boards: 2
     standalone_extension: 0 # By default disabled
+    # Custom dataset section.
+    custom_dataset_query: ''        # Write JQL query to add JQL output to the datasets/jira/custom-issues.csv
```
Comment (ometelytsia) [inline app/jira.yml:38]: app/datasets/jira/custom-issues.csv

Comment (ometelytsia) [inline app/jira.yml:38]: Need to add example

File: app/extension/jira/extension_ui.py [hunk 7]
```diff
@@ -3,6 +3,8 @@
...
+    BoardsList, Board, Dashboard, Logout
 
 
 def app_specific_action(webdriver, datasets):
```
Comment (ometelytsia) [inline app/extension/jira/extension_ui.py:10]: custom dataset usage example has to be added

File: app/extension/confluence/extension_ui.py [hunk 8]
```diff
@@ -3,6 +3,8 @@
...
+    Logout
 
 
 def app_specific_action(webdriver, datasets):
```
Comment (ometelytsia) [inline app/extension/confluence/extension_ui.py:10]: custom dataset usage example has to be added

File: app/selenium_ui/confluence_ui.py [hunk 9]
```diff
@@ -40,6 +40,12 @@ def test_1_selenium_view_dashboard(confluence_webdriver, confluence_datasets, co
 #     extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)
 
 
+def test_1_selenium_custom_view_page(confluence_webdriver, confluence_datasets, confluence_screen_shots):
```
Comment (ometelytsia) [inline app/selenium_ui/confluence_ui.py:43]: what is that?

File: app/selenium_ui/confluence_ui.py [hunk 10]
```diff
@@ -40,6 +40,12 @@ def test_1_selenium_view_dashboard(confluence_webdriver, confluence_datasets, co
...
+    extension_ui.view_page_custom_page(confluence_webdriver, confluence_datasets)
+
+def test_1_selenium_custom_edit_page(confluence_webdriver, confluence_datasets, confluence_screen_shots):
```
Comment (ometelytsia) [inline app/selenium_ui/confluence_ui.py:46]: what is that?

File: app/util/data_preparation/confluence_prepare_data.py [hunk 11]
```diff
@@ -75,6 +79,16 @@ def __get_pages(confluence_api, count):
     return pages
 
 
+def __get_custom_pages(confluence_api, count, cql):
+    pages = []
+    if cql:
+        pages = confluence_api.get_content_search(
+            0, count, cql=cql)  # filter out space Home pages
```
Comment (ometelytsia) [inline app/util/data_preparation/confluence_prepare_data.py:86]: comment is confusing

File: app/util/data_preparation/confluence_prepare_data.py [hunk 12]
```diff
@@ -75,6 +79,16 @@ def __get_pages(confluence_api, count):
...
+        if not pages:
+            raise SystemExit(f"There are no Pages in Confluence could be find with CQL: {cql}")
```
Comment (ometelytsia) [inline app/util/data_preparation/confluence_prepare_data.py:88]: "No pages were found by CQL: {cql}"

File: app/util/data_preparation/jira_prepare_data.py [hunk 13]
```diff
@@ -116,8 +123,19 @@ def __get_issues(jira_api, software_projects):
...
+
 
+def __get_custom_issues(jira_api, custom_jql):
+    issues = []
+    if custom_jql:
+        issues = jira_api.issues_search(
+            jql=custom_jql, max_results=8000
+        )
+        if not issues:
+            print(f"There are no issues found using JQL {custom_jql}")
```
Comment (ometelytsia) [inline app/util/data_preparation/jira_prepare_data.py:138]: Why print, but not SystemExit?

Comment (ometelytsia) [inline app/util/data_preparation/jira_prepare_data.py:138]: "No issues were found by JQL: {custom_jql}"

File: app/extension/confluence/extension_ui.py (hunk 4 above)
Comment (SergeyMoroz0703) [reply app/extension/confluence/extension_ui.py:6]: fixed

File: app/extension/jira/extension_ui.py (hunk 7 above)
Comment (SergeyMoroz0703) [reply app/extension/jira/extension_ui.py:10]: added

**Pull request closed without merging**